.changeset/unlucky-laws-fetch.md

```diff
@@ -0,0 +1,15 @@
+---
+"wrangler": minor
+---
+
+R2 will introduce storage classes soon. Wrangler allows you to interact with storage classes once it is
+enabled on your account.
+
+Wrangler supports an `-s` flag that allows the user to specify a storage class when creating a bucket,
+changing the default storage class of a bucket, and uploading an object.
+
+```bash
+wrangler r2 bucket create ia-bucket -s InfrequentAccess
+wrangler r2 bucket put-storage-class my-bucket -s InfrequentAccess
```

Comment:
Oh from looking at the code it should be:

```suggestion
wrangler r2 bucket put storage-class my-bucket -s InfrequentAccess
```

The extra noun "put" seems a bit off here. The object of the command is the `bucket` still and the operation is `update-storage-class`, no?

Reply:
So I would imagine this command should be:

```suggestion
wrangler r2 bucket update-storage-class my-bucket -s InfrequentAccess
```

Reply:
Ah, sorry -- that's my bad. I updated the command itself, but not the changeset file. The reason for using `put` even though it's a `PATCH` request is because the action itself is called `PutBucketStorageClass` in the API to be consistent with other operations that update a bucket settings (`PutBucketLifecycleConfiguration`, `PutBucketCors`, etc.). So, this is why there's a `put` subcommand as we'd also like to expand Wrangler to include the other bucket actions.

Reply:
Although, looking at this again, I do agree that `update` makes it much more intuitive from a user perspective as the operations **is** a `PATCH` request. Updating this to replace the `put` subcommand with `update` subcommand.